cmd/add.go

```diff
@@ -42,13 +55,75 @@ func runAdd(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	err = lock.AddResource(args, algo, tags, filename)
+	cache, err := cmd.Flags().GetString("cache")
 	if err != nil {
 		return err
 	}
+
+	// Check if ARTIFACTORY_TOKEN is set if cache URL is provided
+	token := os.Getenv("ARTIFACTORY_TOKEN")
+	if token == "" && cache != "" {
+		log.Warn().Msg("ARTIFACTORY_TOKEN is not set; cache upload may fail if authentication is required.")
+	}
+
+	// If it's a local file path and cache URL is provided, upload the file
+	if !isRemote && cache != "" {
+		err := uploadToArtifactory(args[0], cache, token)
+		if err != nil {
+			return fmt.Errorf("failed to upload to Artifactory: %w", err)
+		}
+		log.Info().Msgf("Successfully uploaded resource to Artifactory: %s", cache)
```

Comment:
This should probably be debug.  In most successful cases, grabit has no output / is silent.